src/dotnet/Transcription.Service/Google/GoogleTranscriber.cs

```diff
@@ -4,129 +4,206 @@
     }
 
-    public IAsyncEnumerable<Transcript> Transcribe(
+    public async IAsyncEnumerable<Transcript> Transcribe(
         Symbol transcriberKey,
         string streamId,
         AudioSource audioSource,
         TranscriptionOptions options,
-        CancellationToken cancellationToken)
-    {
-        var recognizerTask = GetRecognizer(options, cancellationToken);
-        var process = new GoogleTranscriberProcess(recognizerTask, streamId, audioSource, options, Log);
-        process.Run().ContinueWith(_ => process.DisposeAsync(), TaskScheduler.Default);
-        return process.GetTranscripts(cancellationToken);
-    }
-
-    // Private methods
-
-    private Task<string> GetRecognizer(TranscriptionOptions options, CancellationToken cancellationToken)
+        [EnumeratorCancellation] CancellationToken cancellationToken)
     {
         var region = GetRegionId(options.Language);
         var languageCode = GetLanguageCode(options.Language);
+        var projectId = await GetProjectId().ConfigureAwait(false);
+        var endpoint = $"{region}-speech.googleapis.com:443";
         var recognizerId = $"{languageCode.ToLowerInvariant()}";
+        var parent = $"projects/{projectId}/locations/{region}";
+        var recognizerName = $"{parent}/recognizers/{recognizerId}";
+        var builder = new SpeechClientBuilder {
+            Endpoint = endpoint,
+        };
+        var speechClient = await builder.BuildAsync(cancellationToken).ConfigureAwait(false);
+        var recognizeRequests = speechClient
+            .StreamingRecognize(
+                new CallSettings(cancellationToken, Expiration.FromTimeout(TimeSpan.FromMinutes(3)), null, null, null, null),
+                new BidirectionalStreamingSettings(1));
+        var streamingRecognitionConfig = new StreamingRecognitionConfig {
+            Config = new RecognitionConfig {
+                AutoDecodingConfig = new AutoDetectDecodingConfig(),
+            },
+            StreamingFeatures = new StreamingRecognitionFeatures {
+                InterimResults = true,
+                // TODO(AK): test google VAD events - probably it might be useful
+                // VoiceActivityTimeout =
+                // EnableVoiceActivityEvents =
+            },
+        };
+        Log.LogDebug("Starting recognize process for {StreamId}", streamId);
+        await recognizeRequests.WriteAsync(new StreamingRecognizeRequest {
+            StreamingConfig = streamingRecognitionConfig,
+            Recognizer = recognizerName,
+        }).ConfigureAwait(false);
+
+        var webMStreamAdapter = new WebMStreamAdapter(Log);
+        var silenceAudio = await _silenceAudioSourceTask.ConfigureAwait(false);
+        var audioSourceJoined = silenceAudio
+            .Take(TimeSpan.FromMilliseconds(2000), cancellationToken)
+            .Concat(audioSource, cancellationToken)
+            .ConcatUntil(silenceAudio, TimeSpan.FromSeconds(7), cancellationToken);
```

Comment:
This overload is actually weird - `TimeSpan.FromSeconds(7)` here reads as total length, I'd probably leave just `Concat` overload with this parameter, assuming we name params there as `this head, tail, maxTailDuration`.

Reply:
Because this param doesn't limit the tail - but the whole stream - so it works similar to PadRight